output/slide-deck.qmd

```diff
@@ -614,37 +535,44 @@ I)](https://blogs.cisco.com/learning/git-for-network-engineers-series-the-basics
 ::: notes
 -   Deleted folders "do not exist". It only considers "deleted files".
 
--   Deleting an empty folders does not show anything in the Git tab.
+-   Deleting an empty folder will not show anything in the Git tab.
 
 -   Deleting a non-empty folder shows one line for each deleted file in the Git
     tab.
 
 -   This is why respositories should NEVER be nested
 
 -   REVERT: Recovers the LAST status of a file from the last point in history.
+
+-   Reverting can be tried with "README.md" (but NOT with "README.Rmd"); then it
+    can be knit again.
 :::
 
 ## Iterating our work
```

Comment:
It might be easier to go through the image/diagram and immediately define all the definitions (more jargon and even more jargon slides) before going into the example. That way, they sorta understand how the whole system works, and then this can collapse the jargon slide into one slide and paired with a more clear image of the git system, might help their overall understanding of why they are doing what their doing/how the system looks on a whole level. I know it's late in the game to be changing this, so I leave this decision up to you!

Reply:
I'm more inclined to drop the "lifecycle" figure altogether (it was intended only as a sort of "section introduction" more than an explanation). I prefer to use an approach of "how to do things", more than "understanding" upfront how the system works, strange as it may seem. Of course, "why we do things" is also important, but I assume (maybe incorrectly) that the attendees will have the motivation to do it this way.

If you want to know more about why I propose this approach, I take these ideas from this book: [The Nurnberg Funnel](https://mitpress.mit.edu/9780262031639/the-nurnberg-funnel/) (that, and the frustration of trying to teach data science the wrong way for many years 😆 ).

Reply:
FWIW, it's one of the reasons why I don't like very much following the Software Carpentry git lesson; it's too focused on understanding the system insides, instead of focusing on satisfying the needs of the trainees... Totally missing the pedagogical principles they teach themselves in the instructor training course! 😆